### ISSUE TITLE
Delegation store should allow listing accounts delegated to me

### ISSUE THREAD

**Issue by quantranhong1999** [linagora/james-project#4673]
## How 
 - Add APIs to `DelegationStore`:
```java
    Publisher<Username> delegatedAccounts(Username baseUser);

    Publisher<Void> removeDelegatedAccount(Username baseUser, Username delegatedToUser);
```

  - Cassandra and memory implementation: in `user` table add a column named `delegated_account` (any better naming idea?) that stores Set of delegated (to me) users
    Data model: Username,.., Set of authorized users, Set of delegated account users

    Remark:
    - when user A add/remove authorized user B, we need to add/remove delegated accounts set of user B accordingly. E.g:
    ```
    User  | authorized users |  delegated accounts
    A         Set(B)                     ...
    B         ...                        Set(A)
    ```

    - when user B remove delegated account A, we need to modify accordingly for user A. E.g:

    ```
    User  | authorized users |  delegated accounts
    A       Set()                ...
    B       ...                  Set()
    ```

##  DoD 
implementation + unit tests

**Comment by chibenwa** [comment]
What is the source of truth: authorized_users or delegated_accounts? Depending of the answer, we could position retries to limit inconsistencies.

Maybe also that could be a good candidate for (logged) batch in order not to have inconsistencies (here we do not really care about the write performances...).

**Comment by chibenwa** [comment]
Have a lenient behaviour for non existing source users.

 - If the source user exist, record the denation for them.
 - If the source user do not exist, do not create the entry (and do not fail)

**Comment by chibenwa** [comment]
We can use (logged) BATCH in Cassandra

**Comment by quantranhong1999** [comment]
@vttranlina please next time comment PRs link in the issue. The GitHub board does not show "Connected to PR blabla..." so it is hard to follow.

**Comment by vttranlina** [comment]
> @vttranlina please next time comment PRs link in the issue. The GitHub board does not show "Connected to PR blabla..." so it is hard to follow.

Your account don't show it?
![image](https://user-images.githubusercontent.com/81145350/210596311-192acfb0-f813-4d12-b481-318338b138ac.png)
it show only me?

**Comment by quantranhong1999** [comment]
> it show only me? 

I mean the new Github board, not (opening a new tab for) the Github issue to see the connected PR.
That would be more convenient to me.

**Comment by vttranlina** [comment]
PR: https://github.com/apache/james-project/pull/1372
